# HG changeset patch
# User sbr <sebastien@brissaud.name>
# Date 1370378692 -7200
# Branch fs5567_02
# Node ID 5874f43c93969aa6acdfe016e708492e5b555c9f
# Parent  4b2f6a972a32ff529e71228d0b8dcb2d24aa9dfe
# Parent  4ce6f9e1d3f07e01750d6ac5449dab55a03926d8
-Fix [FS#5567] (r25283): Use the UI index of the selected object to make it visible when re-opening the build object window.

diff --git a/src/object_gui.cpp b/src/object_gui.cpp
--- a/src/object_gui.cpp
+++ b/src/object_gui.cpp
@@ -109,7 +109,7 @@
 			this->FindWindowPlacementAndResize(this->width, this->height);
 		}
 
-		if (_selected_object_index != -1) matrix->SetClicked(_selected_object_index);
+		if (_selected_object_index != -1) matrix->SetClicked(ObjectClass::Get(_selected_object_class)->GetUIFromIndex(_selected_object_index));
 	}
 
 	virtual ~BuildObjectWindow()
